Update dev docs and rzup readme to use revamed rzup (#2202)
risc0/risc0: website/api/zkvm/install.md

See `rzup --help` for more options. You can find out more about `rzup` [here](https://github.com/risc0/risc0/tree/main/rzup).

### Installation for all other systems (e.g. x86-64 macOS, arm64 Linux)

> For x86-64 macOS or arm64 linux, you must build `rzup` from source. Clone the repository by running `git clone https://github.com/risc0/risc0.git`. Go to the root of the repository and run `cargo install --path rzup`. Build and install the `cargo-risczero` by first checking out the branch `release-*` where `*` is `[major release number].[minor release number]` of your desired zkVM version. For example, if you would like to install version 1.1.0, run `git checkout origin/release-1.1` and run `cargo install --path risc0/cargo-risczero`. Build and install the rust toolchain by running `rzup toolchain build rust` this command may require utilities such as `cmake` and the `ninja` build system to be installed.

Reviewer: This is hard to read, can we split this out into steps?

Reviewer: I also think this should go down into Manual Installation

> At this time, there is only one release of the `cpp` toolchain. Installing the default version will fall back to this release.

> 2. For x86-64 macOS or arm64 linux, you must build `rzup` from source. Clone the repository by running `git clone https://github.com/risc0/risc0.git`. Go to the root of the repository and run `cargo install --path rzup`. Build and install the rust toolchain by running `rzup toolchain build rust`.

Reviewer: ```suggestion
2. For all other systems (e.g. macOS/x86-64, Linux/arm64), you must build `rzup` from source. 
* Clone the repository with: `git clone https://github.com/risc0/risc0.git`. 
* In the root of the repository, install `rzup` with: `cargo install --path rzup`.
* Build and install the rust toolchain with: `rzup toolchain build rust`.
```

Author: suggestion applied!
